### PR TITLE
CLOUDP-155672: Fixing assignment for defaultRWConcern configs

### PR DESCRIPTION
**Pull request by colm-quinn**
Fixing issue where DefaultRWConcern was being dropped from mongocli cluster describe/update

Example:
```
mongocli om cluster describe colm-replset-testing --output json
{
  "name": "colm-replset-testing",
  "processes": [
    {
      "arbiterOnly": false,
      "buildIndexes": true,
      "dbPath": "/data/colm-replset-testing",
      "defaultRWConcern": {},
      "disabled": false,
      "featureCompatibilityVersion": "6.0",
      "hidden": false,
      "hostname": "{redacted}",
      "logDestination": "file",
      "logPath": "/data/colm-replset-testing/mongodb.log",
      "name": "colm-replset-testing_31",
      "port": 25000,
      "priority": 1,
      "processType": "mongod",
      "secondaryDelaySecs": 0,
      "votes": 1,
      "version": "6.0.2"
    },
    {
      "arbiterOnly": false,
      "buildIndexes": true,
      "dbPath": "/data/colm-replset-testing/rc1",
      "defaultRWConcern": {},
      "disabled": false,
      "featureCompatibilityVersion": "6.0",
      "hidden": false,
      "hostname": "{redacted}",
      "logDestination": "file",
      "logPath": "/data/colm-replset-testing/rc1/mongodb.log",
      "name": "colm-replset-testing_32",
      "port": 25001,
      "priority": 1,
      "processType": "mongod",
      "secondaryDelaySecs": 0,
      "votes": 1,
      "version": "6.0.2"
    }
  ],
  "mongoURI": "{redacted}"
}
```

Old update:
```
mongocli om cluster describe colm-replset-testing
Error: GET {redacted}: 400 (request "PATH_PARAM_PARSE_ERROR") One or more path parameter in the request URI /api/public/v1.0/groups/5e443a1253de6a195497afdd/clusters/colm-replset-testing is malformed.
```

New update:
```
mongocli om cluster update colm-replset-testing -f repl-set-update.json
Changes are being applied, please check {redacted}/v2/5e443a1253de6a195497afdd#deployment/topology for status
```

Closes #[HELP-43182]

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

**Review by wtrocki — APPROVED** [review]
LGTM (did not verified changes)

File: internal/convert/process_config.go [hunk 1]
```diff
@@ -485,6 +485,7 @@ func (p *ProcessConfig) process() *opsmngr.Process {
 	process := &opsmngr.Process{
 		AuthSchemaVersion:                authSchemaVersion,
 		BackupRestoreCheckpointTimestamp: p.BackupRestoreCheckpointTimestamp,
+		DefaultRWConcern:                 p.defaultRW(),
```
Comment (andreaangiolillo) [inline internal/convert/process_config.go:488]: is this the main fix? Could you explain why this is needed? I see that `&opsmngr.Process` has many other properties where we don't set the default value and I am not sure why here is needed.

File: internal/convert/cluster_config_test.go [hunk 2]
```diff
@@ -27,6 +27,7 @@ import (
 
 func TestClusterConfig_PatchAutomationConfig(t *testing.T) {
 	fipsMode := true
+	defaultWriteConcernJ := true
```
Comment (andreaangiolillo) [inline internal/convert/cluster_config_test.go:30]: it would be great to update the e2e test to run an update command [e2e/cloud_manager/deploy_replica_set_test.go](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/e2e/cloud_manager/deploy_replica_set_test.go). Happy to move it to another PR.

**Review by andreaangiolillo — APPROVED** [review]
My comments do not block the PR. LGTM

File: internal/convert/process_config.go (hunk 1 above)
Comment (colm-quinn) [reply internal/convert/process_config.go:488]: Side-effect observed rather than the main change.

By default, the DefaultRWConcern isn't specified and can be safely ignored - it only becomes an issue if the user has updated the setting at some point. Once the value has been set once, it needs to be present in all further config updates, otherwise the config is rejected as the existing value would be changed to nil.  

Regarding the other settings, they would all need to be tested individually to see the impact. Currently, the mongocli cluster update command only covers a subset of allowed settings.

File: internal/convert/cluster_config_test.go (hunk 2 above)
Comment (colm-quinn) [reply internal/convert/cluster_config_test.go:30]: Good idea. Capturing in https://jira.mongodb.org/browse/CLOUDP-166012 for triage.
